javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typeinference/LeastUpperBoundLogic.java

```diff
@@ -276,6 +276,8 @@ private List<Set<ResolvedType>> supertypes(Set<ResolvedType> types) {
 
     private Set<ResolvedType> supertypes(ResolvedType type) {
         // How to deal with other types like for example type variable?
+        if (type.isTypeVariable())
```

Comment:
LeatUpperBoundLogic does not currently support generic types, in particular the notion of inheritance. This PR does not solve this shortcoming but it may be a workaround for the issue you raised.